app/presenters/vendor_api/single_application_presenter.rb

```diff
@@ -255,6 +261,59 @@ def qualification_to_hash(qualification)
       }.merge HesaQualificationFieldsPresenter.new(qualification).to_hash
     end
 
+    def subject_code(qualification)
+      # There are a few cases here:
+      #
+      # Qualification level: gcse
+      #   These are either maths, science or english GCSEs that are required by some courses
+      #   - Maths GCSEs are not structured, so we map to a single maths subject code
+      #   - Science GCSEs are structured, but we combine them into one in the API response, so we can only be as granular as 'science'
+      #   - English GCSEs are now structured, but they did not used to be. We map to a generic english code here for the unstructured ones,
+      #     and have a better mapping within the structured_gcse_to_hashes method for newer english GCSEs
+      #
+      # Qualification level: other
+      #   These are other qualifications that a candidate has that are not direct requirements for the course.
+      #   This includes GCSEs and A levels, and none of them are structured. If their subject has been chosen
+      #   directly from the autocomplete, then we can map to a code. Otherwise, we leave the code blank.
+
+      if qualification.gcse?
+        subject_code_for(qualification.subject)
+      elsif qualification.other?
+        subject_code_for_other_qualification(qualification)
+      end
+    end
+
+    def subject_code_for(subject)
+      case subject
```

Comment:
Im quite unsure about the different subjects and mappings here, let's have a chat in person. 

But my main concern is this logic should not live here, and be extracted into the `ApplicationQualification` and `A_AND_AS_LEVEL_SUBJECTS_TO_CODES` etc.